ddev/src/ddev/cli/application.py

```diff
@@ -87,8 +87,11 @@ def initialize_old_cli(self):
         self.__config['color'] = not self.console.no_color
         self.__config['dd_api_key'] = self.config.orgs.get('default', {}).get('api_key', '')
         self.__config['dd_app_key'] = self.config.orgs.get('default', {}).get('app_key', '')
-
-        kwargs = {'here' if self.repo.name == 'local' else self.repo.name: True}
+        # Make sure that envvar overrides of repo make it into config.
+        self.__config['repo'] = self.repo.name
+        kwargs = {}
+        if self.repo.name == 'local':
+            kwargs['here'] = True
```

Comment:
At first sight, I have no idea what it's doing and why. Could you add more info to the description?

Reply:
Thanks, added an explanation. Needed a sec to dig it up myself, which is a good indicator that we need the comment!